DNN Platform/Library/Services/GeneratedImage/ProfileEventHandler.cs

```diff
@@ -24,9 +24,11 @@ public void ProfileUpdated(object sender, ProfileEventArgs profileArgs)
             if (profileArgs?.User == null || profileArgs.OldProfile == null) return;
             //extract old and new user profile from args and clear both client and server caching
             var user = profileArgs.User;
+            var newProfile = user.Profile;
             var oldProfile = profileArgs.OldProfile;
-            if (user.Profile.Photo != oldProfile.Photo || user.Profile.GetProperty("Photo").ProfileVisibility.VisibilityMode !=
-                oldProfile.ProfileProperties["Photo"].ProfileVisibility.VisibilityMode)
+            var newPhotoVisibilityMode = newProfile.GetProperty("Photo")?.ProfileVisibility.VisibilityMode;
+            var oldPhotoVisibilityMode = oldProfile.GetProperty("Photo")?.ProfileVisibility.VisibilityMode;
```

Comment:
Please replace the property name with `Entities.Users.UserProfile.USERPROFILE_Photo` constant.